### PR TITLE
Feature/update to 3 validators

### PR DESCRIPTION
**Pull request by Uh-X3L**
Here’s a Pull Request description you can use:

🚀 Upgrade Validator Selection to 3 Validators
Overview
This PR enhances the PrediFi protocol by upgrading the validator selection mechanism from 2 to 3 validators. This change addresses previous issues with ties and lapses in the validation process, making the protocol more robust and reliable.

✨ What’s Changed
Validator Selection:
Updated the logic to randomly select 3 unique validators for each pool instead of 2.
State & Interface Updates:
Modified contract state and interfaces to store and return 3 validator addresses per pool.
Event Emission:
Updated the ValidatorsAssigned event to emit 3 validator addresses.
Unit Tests:
Added and updated tests to ensure correct assignment and uniqueness of 3 validators, including edge cases.
Error Handling:
Improved checks to prevent duplicate assignments and ensure sufficient validators are available.
Documentation:
Updated relevant docstrings and comments to reflect the new 3-validator system.
🧪 Testing
All tests pass using the standard workflow.

### CODE REVIEW

File: contract/src/predifi.cairo [hunk 1]
```diff
@@ -685,59 +686,71 @@ pub mod Predifi {
...
             } else if validator_count == 1 {
-                // Only one validator available, assign the same validator twice
                 let validator = self.validators.at(0).read();
-                self.assign_validators(pool_id, validator, validator);
+                self.assign_validators(pool_id, validator, validator, validator);
+                return;
+            } else if validator_count == 2 {
+                let validator1 = self.validators.at(0).read();
+                let validator2 = self.validators.at(1).read();
+                self.assign_validators(pool_id, validator1, validator2, validator1);
```
Comment (Akshola00) [inline contract/src/predifi.cairo:707]: why are we still using validator one here?

**Review by Akshola00 — COMMENTED** [review]
apart from this the upgrade is alright

**Comment by Akshola00** [comment]
ci is fialing, run scarb fmt

File: contract/src/predifi.cairo (hunk 1 above)
Comment (Uh-X3L) [reply contract/src/predifi.cairo:707]: @Akshola00  Explained in detail in twitter. It covers the edge case where you have only 2 validators to choose from, above this there was also the edge case with only 1 validator to choose from. So I added this in order to keep with the coding style and amplifying the coverage of the test to +1 validators.

**Comment by Uh-X3L** [comment]
@martinvibes  @Akshola00  Created a new issue in order to fix the test to avoid deterministic behaviour:
https://github.com/Web3Novalabs/predifi/issues/147
